Assets/BossRoom/Scripts/Shared/Game/Entity/NetworkCharacterState.cs

```diff
@@ -113,7 +113,7 @@ public LifeState LifeState
         public override void OnNetworkSpawn()
         {
             if (!IsServer) return;
-            HitPoints = CharacterClass.BaseHP.Value;
+            if (HitPoints == 0) HitPoints = CharacterClass.BaseHP.Value;
         }
```

Comment:
We should have the health initialized at the same place in code. Since we're doing that now for the late-joined player inside ServerBossRoomState, we might as well just do this there for a fresh player.

Reply:
I thought about it too, but this code is used for NPCs as well. So this is needed for them. I'll try to see if I can do the reverse and get the info from the SessionManager here and remove the setting of HP in ServerBossRoomState

Reply:
That's very true. Hopefully that works out though.